crates/ruff_cli/src/lib.rs

```diff
@@ -43,6 +43,42 @@ impl From<ExitStatus> for ExitCode {
     }
 }
 
+#[derive(Default)]
+pub struct FileChangeDetector<'a> {
+    extensions: Vec<&'a str>,
+    config_files: Vec<&'a str>,
+    has_configuration_changed: bool,
+}
+
+impl FileChangeDetector<'_> {
```

Comment:
We can keep this a standalone function and instead introduce a new tri-boolean type to signal whether to ignore the change, it's a python file, or a configuration change

```rust
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
enum ChangeKind {
	Configuration,
	SourceFile,
	Ignore
}

fn detect_change(paths: &[PathBuf]) -> ChangeKind {
	...
}
```